src/lib.rs

```diff
@@ -71,15 +71,60 @@ impl WaiterChannel {
     }
 }
 
-struct Scheduler {
-    waiter: Arc<WaiterChannel>,
-    heap: BinaryHeap<Schedule>,
+/// A trait that allows configurable execution of scheduled item
+/// on the scheduler thread.
+trait Executor<T> {
+    fn execute(&mut self, data : T);
```

Comment:
I agree, this was very unfortunate. Box<FnMut()> just behaves so differently than other types that I wasn't sure what the proper solution here was. Box<FnMut()> requires a mutable reference to execute, and yet we don't want to clone the box for every subsequent callback invocation.

Only having "fn execute(&mut self, data : &mut T)" would require MessageTimer to clone every message, regardless of whether it was required or not (since the cloning would happen within execute).

Only having "fn execute(&mut self, data : T)" would require Timer to clone Box<FnMut()> instances whenever repeats are scheduled.

Do you have any ideas on a better signature that has the proper performance characteristics for these types? If not, can we address this by adding further comments?